include/deal.II/dofs/dof_handler.h

```diff
@@ -1805,6 +1845,9 @@ class DoFHandler : public Subscriptor
   // in the functions that set and retrieve vertex dof indices.
   friend struct dealii::internal::DoFHandlerImplementation::Implementation;
   friend struct dealii::internal::hp::DoFHandlerImplementation::Implementation;
+  friend unsigned int
+  dealii::internal::hp::DoFHandlerImplementation::
+    dominated_future_fe_on_children<dim, spacedim>(const cell_iterator &);
```

Comment:
Would it work if we added this as another static function inside `hp::DoFHandlerImplementation::Implementation` so that we don't need another `friend` declaration?

Reply:
We not necessarily need access to private members for the function in the current state, but I most certainly will when preparing the fix I have in mind for #11902.

Hmm. Would it work if I move this particular function into the `Implementation` struct, and then basically call a wrapper that is part of the public interface? I'll give it a try!